File: shared/Models/Side.swift
import Foundation

/// Logical team side: left = Us, right = Them. Visual layout may remap when swap-sides is on.
public enum Side: String, Codable, Sendable, CaseIterable, Equatable {
    case left
    case right

    public var opposite: Side {
        switch self {
        case .left: return .right
        case .right: return .left
        }
    }

    public var displayName: String {
        switch self {
        case .left: return "Us"
        case .right: return "Them"
        }
    }
}

File: android/domain/src/main/kotlin/com/wristrally/domain/Side.kt
package com.wristrally.domain

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Logical team side: left = Us, right = Them. Visual layout may remap when swap-sides is on. */
@Serializable
enum class Side {
    @SerialName("left")
    Left,

    @SerialName("right")
    Right,
    ;

    val opposite: Side
        get() = if (this == Left) Right else Left

    val displayName: String
        get() = if (this == Left) "Us" else "Them"
}
